src/main/java/org/terasology/las/platform/FloatingPlatformProvider.java

```diff
@@ -77,7 +83,15 @@ public void process(GeneratingRegion region) {
                 platformFacet.add(platform);
             }
         }
-
+        BlockRegion worldRegion = surfacesFacet.getWorldRegion();
+        for (int x = worldRegion.minX(); x <= worldRegion.maxX(); x++) {
+            for (int z = worldRegion.minZ(); z <= worldRegion.maxZ(); z++) {
+                int y = surfacesFacet.getNextBelow(new Vector3i(x, LASUtils.FLOATING_PLATFORM_HEIGHT_LEVEL, z));
+                if (worldRegion.contains(x, y, z) && FLOATING_PLATFORM_REGION.contains(x, z)) {
+                    surfacesFacet.setWorld(x, y, z, false);
+                }
+            }
+        }
```

Comment:
This creates a new `Vector3i` each time, which is unnecessarily expensive. It would already be a lot better if we'd reuse the same temporary vector.

```suggestion
        Vector3i pos = new Vector3i();
        for (int x = worldRegion.minX(); x <= worldRegion.maxX(); x++) {
            for (int z = worldRegion.minZ(); z <= worldRegion.maxZ(); z++) {
                pos.set(x, LASUtils.FLOATING_PLATFORM_HEIGHT_LEVEL, z);
                int y = surfacesFacet.getNextBelow(pos);
                if (worldRegion.contains(x, y, z) && FLOATING_PLATFORM_REGION.contains(x, z)) {
                    surfacesFacet.setWorld(x, y, z, false);
                }
            }
        }
```

However, it would be even better to simply add the respective variants taking three integer arguments to the `SurfacesFacet`. The values are "unpacked" in those methods anyway, so I think this is reasonable addition.

```suggestion
        for (int x = worldRegion.minX(); x <= worldRegion.maxX(); x++) {
            for (int z = worldRegion.minZ(); z <= worldRegion.maxZ(); z++) {
                int y = surfacesFacet.getNextBelow(x, LASUtils.FLOATING_PLATFORM_HEIGHT_LEVEL, z);
                if (worldRegion.contains(x, y, z) && FLOATING_PLATFORM_REGION.contains(x, z)) {
                    surfacesFacet.setWorld(x, y, z, false);
                }
            }
        }
```

Reply:
looks like it doesn't get unpacked as the only argument to that function is a vector.